<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsCompositionWidgetBase</class>
 <widget class="QWidget" name="QgsCompositionWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>416</width>
    <height>620</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Composition</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <property name="margin">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-72</y>
        <width>396</width>
        <height>688</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Paper and quality</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="textLabel3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Size</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mPaperSizeComboBox</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QComboBox" name="mPaperSizeComboBox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="textLabel5">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Units</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mPaperUnitsComboBox</cstring>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QComboBox" name="mPaperUnitsComboBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="textLabel4">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Width</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mPaperWidthLineEdit</cstring>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLineEdit" name="mPaperWidthLineEdit"/>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="textLabel6">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Height</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mPaperHeightLineEdit</cstring>
            </property>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLineEdit" name="mPaperHeightLineEdit"/>
          </item>
          <item row="8" column="0">
           <widget class="QLabel" name="textLabel7">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Orientation</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mPaperOrientationComboBox</cstring>
            </property>
           </widget>
          </item>
          <item row="9" column="0">
           <widget class="QComboBox" name="mPaperOrientationComboBox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QLabel" name="mResolutionLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Print quality (dpi)</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mResolutionLineEdit</cstring>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLineEdit" name="mResolutionLineEdit"/>
          </item>
          <item row="12" column="0">
           <widget class="QCheckBox" name="mPrintAsRasterCheckBox">
            <property name="text">
             <string>Print as raster</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="mSnapGroupBox">
         <property name="title">
          <string>Snapping</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QCheckBox" name="mSnapToGridCheckBox">
            <property name="text">
             <string>Snap to grid</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="mResolutionLabel_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Grid resolution</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mResolutionSpinBox</cstring>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QDoubleSpinBox" name="mResolutionSpinBox"/>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="mXOffsetLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Offset x</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mOffsetXSpinBox</cstring>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QDoubleSpinBox" name="mOffsetXSpinBox"/>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="mYOffsetLabel">
            <property name="text">
             <string>Offset y</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mOffsetYSpinBox</cstring>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QDoubleSpinBox" name="mOffsetYSpinBox"/>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="mPenWidthLabel">
            <property name="text">
             <string>Pen width</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mPenWidthSpinBox</cstring>
            </property>
           </widget>
          </item>
          <item row="8" column="0">
           <widget class="QDoubleSpinBox" name="mPenWidthSpinBox"/>
          </item>
          <item row="9" column="0">
           <widget class="QLabel" name="mGridColorLabel">
            <property name="text">
             <string>Grid color</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mGridColorButton</cstring>
            </property>
           </widget>
          </item>
          <item row="10" column="0">
           <widget class="QgsColorButton" name="mGridColorButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="11" column="0">
           <widget class="QLabel" name="mGridStyleLabel">
            <property name="text">
             <string>Grid style</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
            <property name="buddy">
             <cstring>mGridStyleComboBox</cstring>
            </property>
           </widget>
          </item>
          <item row="12" column="0">
           <widget class="QComboBox" name="mGridStyleComboBox"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QgsColorButton</class>
   <extends>QToolButton</extends>
   <header>qgscolorbutton.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>scrollArea</tabstop>
  <tabstop>mPaperSizeComboBox</tabstop>
  <tabstop>mPaperUnitsComboBox</tabstop>
  <tabstop>mPaperWidthLineEdit</tabstop>
  <tabstop>mPaperHeightLineEdit</tabstop>
  <tabstop>mPaperOrientationComboBox</tabstop>
  <tabstop>mResolutionLineEdit</tabstop>
  <tabstop>mPrintAsRasterCheckBox</tabstop>
  <tabstop>mSnapToGridCheckBox</tabstop>
  <tabstop>mResolutionSpinBox</tabstop>
  <tabstop>mOffsetXSpinBox</tabstop>
  <tabstop>mOffsetYSpinBox</tabstop>
  <tabstop>mPenWidthSpinBox</tabstop>
  <tabstop>mGridColorButton</tabstop>
  <tabstop>mGridStyleComboBox</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
